### PR TITLE
Remove ability for plugins to register a MIR pass

### PR DESCRIPTION
**Pull request by nagisa**
In recent months there have been a few different people investigating how to make a plugin that
registers a MIR-pass – one that isn’t intended to be eventually merged into rustc proper.

The interface to register MIR passes was added primarily for miri (& later was
found to make prototyping of rustc-proper MIR passes a tiny bit faster). Since miri does not use
this interface anymore it seems like a good time to remove this "feature".

For prototyping purposes a similar interface can be added by developers themselves in their custom
rustc build.

cc @nikomatsakis

### CODE REVIEW

File: src/test/run-pass-fulldeps/mir-pass.rs [hunk 1]
```diff
@@ -1,23 +0,0 @@
...
-// file at the top-level directory of this distribution and at
-// http://rust-lang.org/COPYRIGHT.
-//
-// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
-// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
-// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
-// option. This file may not be copied, modified, or distributed
-// except according to those terms.
-
-// aux-build:dummy_mir_pass.rs
```
Comment (rkruppe) [inline src/test/run-pass-fulldeps/mir-pass.rs]: That file should be removed too, I guess?

Comment (nagisa) [inline src/test/run-pass-fulldeps/mir-pass.rs]: Yes.

**Pull request merged**